etc/calc_release_version.py

```diff
@@ -144,27 +144,24 @@ def main():
                                 + release_branch_match.group('brname') + '"')
 
     else:
-        tags = check_output(['git', 'tag',
-                                        '--merged', 'HEAD',
-                                        '--list', 'r*',
-                                        '--sort', 'version:refname'])
-        if len(tags) > 0:
-            release_tag_match = RELEASE_TAG_RE.match(tags.splitlines()[-1].decode('utf-8'))
-            if release_tag_match:
-                version_new = {}
-                version_new['major'] = int(release_tag_match.group('vermaj'))
-                version_new['minor'] = int(release_tag_match.group('vermin'))
-                version_new['patch'] = int(release_tag_match.group('verpatch')) + 1
-                version_new['prerelease'] = prerelease_marker
-                new_version_loose = LooseVersion(str(version_new['major']) + '.' +
-                                                 str(version_new['minor']) + '.' +
-                                                 str(version_new['patch']) + '-' +
-                                                 version_new['prerelease'])
-                if new_version_loose > version_loose:
-                    version_loose = new_version_loose
-                    if DEBUG:
-                        print('Found new best version "' + str(version_loose) \
-                                + '" from tag "' + release_tag_match.group('ver') + '"')
+        tags = check_output (['git', 'describe', '--tags', '--match', 'r*', '--abbrev=0'])
```

Comment:
The version of git (1.9.1) on Ubuntu 14.04 does not support `--merged`. @rcsanchez97 I think this should be equivalent behavior.

Reply:
Yeah, it looks to me like it should be equivalent.  I tried to recall why we specifically chose this approach.  My recollection is vague, but it likely had to do with some possible weirdness associated with branching.  That said, I don't know of a single instance where we have encountered a configuration that would not work with the `describe` command.  I tried a few scenarios by hand locally and there was no unusual or unexpected output.

Reply:
I did the same, it seems equivalent. `git describe` walks backwards to the nearest matching tag. So it gives the same result from `master` and a release branch.

Like before, the result for non-master non-release branches is a little off. I branched directly from master and running gives `3.6.1-20210129+git97cf47e743` when it should probably be `3.7.0-20210129+git97cf47e743`. But... a problem for when we tackle CDRIVER-3315.